src/Cli/dotnet/Commands/Run/VirtualProjectBuildingCommand.cs

```diff
@@ -192,14 +197,28 @@ ProjectRootElement CreateProjectRootElement(ProjectCollection projectCollection)
             projectRoot.FullPath = projectFileFullPath;
             return projectRoot;
         }
+
+        static string GetArtifactsPath(string entryPointFilePath)
+        {
+            // We want a location where permissions are expected to be restricted to the current user.
+            var directory = RuntimeInformation.IsOSPlatform(OSPlatform.Windows)
+                ? Path.GetTempPath()
+                : Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData);
```

Comment:
Do we want to think about cleanup routines for when we're not on Windows using temp? Perhaps it could be a specific option to the run command so that every N runs we do a fire & forget process launch of e.g. `dotnet run --clean-run-cache` so that it doesn't run inside the process the user invoked.

Reply:
Cleaning this automatically would hurt perf because the build artifacts are used for incremental builds (and will be even more when we have caching). But I'd say `dotnet clean` could be updated to clean up this location as well.

Reply:
That challenge with using `dotnet clean` is that it's specific project-centric (you run it on a specific project/solution), whereas in this case I think we'd want more of a cache-centric approach where it cleans up all items using an LRU approach.

Reply:
I think it's similar to the compiler server in that we likely don't have a point where user will realistically run a general `clean-fbp` tool or point `clean` at particular file-based program(s) whose artifacts should be cleaned up. Cleaning on startup feels plausible to me, whether it is done by leaving the N most recent builds in place or builds which are less than a week or a month old or some other basis.

Reply:
Many of these FBPs may be ephemeral. User makes the file, does some experimental thing, then they’re done with it and the file is gone, so it can’t even be passed to dotnet clean.

Reply:
I think we're saying similar things, just that I think we'd want to do it in a way that doesn't impact the run time of an individual invocation, hence my original comment about making it a fire and forget launch of the clean operation on every N runs.

Reply:
Makes sense, but this is something we want to do in a separate PR, right?

Reply:
Clean can come later

Reply:
let's get an issue logged

Reply:
https://github.com/dotnet/sdk/issues/48250